src/marqo/s2_inference/clip_utils.py

```
@@ -156,15 +205,129 @@ def __init__(self, model_type: str = "ViT-B/32", device: str = 'cpu',  embedding
 
-        # https://github.com/openai/CLIP/issues/30
-        self.model, self.preprocess = clip.load(self.model_type, device='cpu', jit=False)
-        self.model = self.model.to(self.device)
-        self.tokenizer = clip.tokenize
-        self.model.eval()
-    
+        path = self.model_properties.get("localpath", None) or self.model_properties.get("url", None)
+
+        if path is None:
+            # The original method to load the openai clip model
+            # https://github.com/openai/CLIP/issues/30
+            self.model, self.preprocess = clip.load(self.model_type, device='cpu', jit=False)
+            self.model = self.model.to(self.device)
+            self.tokenizer = clip.tokenize
+        else:
+            logger.info("Detecting custom clip model path. We use generic clip model loading.")
+            if os.path.isfile(path):
+                self.model_path = path
+            elif validators.url(path):
+                self.model_path = download_pretrained_from_url(path)
+            else:
+                raise InvalidModelPropertiesError(f"The provided model path {path} is neither a local file nor a valid url.")
+
+            self.jit = self.model_properties.get("jit", False)
+            self.device = self.model_properties.get("device", "cpu")
+            self.mean = self.model_properties.get("mean", None)
+            self.std = self.model_properties.get("std", None)
+
+
+            self.model, self.preprocess = self.custom_clip_load()
+            self.tokenizer = self.load_tokenizer()
+
+            self.model.eval()
+
+
+
+    def custom_clip_load(self):
+        self.model_name = self.model_properties.get("name", None)
+
+        if self.model_name in _PRETRAINED:
+            logger.info(f"The name of the custom clip model is {self.model_name}.")
+            model, _, preprocess = open_clip.create_model_and_transforms(model_name=self.model_name, pretrained=self.model_path)
+            return model, preprocess
+
+        else:
```

@jn2clark 

It this "loading from script file" feature  needed?

Should we just throw an error if the `model_name` is not supported?